src/main.rs

```diff
@@ -238,13 +325,73 @@ fn main() -> NazaraResult<()> {
     // Welcome Message.
     println!("{ASCII_ART}");
 
-    // Collect machine information.
-    let machine = Machine {
-        name: None,
-        dmi_information: dmi::construct_dmi_information()?,
-        network_information: network::construct_network_information()?,
-        custom_information: Some(execute(args.plugin)?),
-    };
+    match &args.command {
+        Commands::WriteConfig {
+            uri,
+            token,
+            name,
+            description,
+            comments,
+            status,
+            device_type,
+            site,
+            role,
+            cluster_id,
+            json,
+        } => {
+            println!("Writing configuration file...");
+            if let Some(json_str) = json {
+                write_config_file(
+                    "", // ignored in JSON mode
+                    "", // ignored in JSON mode
+                    name.clone(),
+                    description.clone(),
+                    comments.clone(),
+                    status.clone(),
+                    device_type.clone(),
+                    role.clone(),
+                    site.clone(),
+                    cluster_id.clone(),
+                    Some(json_str.clone()),
```

Comment:
This makes an unnecessary copy of `json`.
```suggestion
                    json,
```